readme.md

```diff
@@ -61,6 +46,38 @@ Repository for both **NEST** and **Elasticsearch.Net**, the two official [Elasti
     </tr>
 </table>
 
+### Further Compatibility Clarifications
+
+#### Can I use a `7.0.0` client against a `5.6` server (new against old), or a `6.0` client against a `7.2` server (old against new)?
+
+No, this is not recommended.
+
+No compatibility assurances are given between different major versions of the client and server. Major differences likely exist between major versions of the server, particularly around request and response object formats.
+
+#### Can I use a `7.0.0` client against a `7.2.0` server?
+
+Both the server and the client are developed according to [SemVer](https://semver.org/) principles, so there should be no breaking changes in a minor version. A `7.0.0` client will work against a `7.2.0` server.
+
+This can be illustrated with some examples;
+
+*Using a `7.0.0` client against a `7.0.0` server*. There is client support for all features in the server. In some instances, we may decide to delay implementing new APIs or some server features until later minor versions of the client, but this is the exception and not the rule.
+
+*Using a `7.0.0` client against a `7.0.1` - `7.2.0` server*. There is client support for all comparable features that are available in the `7.0.0` server. If you want to use new features introduced in `7.2.0` with a lower version client, then you will have to use the low level client `DoRequest` method and perform the request/response (de)serialisation yourself.
+
+When we release a client we will run the unit and integration tests against the latest minor patch version of Elasticsearch that matches the major.minor of the client release, and then any other latest minors within that major version.
+
+Any incompatibilities between minor versions are documented against the release.
+
+#### I have a `6.0` server, what client should I use?
+
+Always use the latest minor version of the client within that major version, so in this instance version `6.8`. The reason being is that `6.8` will contain many bug fixes not present in the `6.0` version of the client.
+
+#### Branch Compatibility
+
+`master` reflects the latest server version (`8.x` snapshots)
+`7.x` reflects the latest development of the `7.x` client
+`7.1` is the integration branch for the next client release (`7.1.0`)
```

Comment:
I would be inclined to keep ephemeral minor integration branches out of the README

Reply:
Could we clarify "at time of writing" as before?

Reply:
This will require updating with each minor release - is that something we want to do?

Reply:
```
`master` reflects the latest server version, this is typically the `current latest major + 1`
`N.x` where N represents the major version component of the Elasticsearch server release its integrating with e,g `7.x`
`N.Y` where `N` is the major version and `Y` is  the minor component, typically opened as integration branch for a specific minor leaving `N.x` free to do bug fix releases.
```

This is the reverse to what Elasticsearch does though, propose we keep `N.x` as the latest `minor` integration branch and open a `N.Y` branch to only backport issues that warrant a bug fix release. We can delete any `< N.Y` branches when we create a new one.

Reply:
This might need duplicating in the `CONTRIBUTING.md`